File: solutions/haiku-1/haiku/go/transpose.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	// Read the first line with R and C
	scanner.Scan()
	parts := strings.Fields(scanner.Text())
	R := atoi(parts[0])
	C := atoi(parts[1])

	// Read the matrix
	matrix := make([][]int, R)
	for i := 0; i < R; i++ {
		scanner.Scan()
		nums := strings.Fields(scanner.Text())
		matrix[i] = make([]int, C)
		for j := 0; j < C; j++ {
			matrix[i][j] = atoi(nums[j])
		}
	}

	// Output the transposed matrix
	for j := 0; j < C; j++ {
		for i := 0; i < R; i++ {
			if i > 0 {
				fmt.Print(" ")
			}
			fmt.Print(matrix[i][j])
		}
		fmt.Println()
	}
}

func atoi(s string) int {
	n, _ := strconv.Atoi(s)
	return n
}

File: solutions/haiku-1/haiku/java/transpose.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int R = sc.nextInt();
        int C = sc.nextInt();

        // Read the matrix
        int[][] matrix = new int[R][C];
        for (int i = 0; i < R; i++) {
            for (int j = 0; j < C; j++) {
                matrix[i][j] = sc.nextInt();
            }
        }

        // Output the transposed matrix
        for (int j = 0; j < C; j++) {
            for (int i = 0; i < R; i++) {
                if (i > 0) System.out.print(" ");
                System.out.print(matrix[i][j]);
            }
            System.out.println();
        }
    }
}
